### PR TITLE
703 feedback on tom swift module from swift operations team

### LINKED ISSUE #10: Swift module not compatible with checking observation status
After installing `tom-swift`, when I run `manage.py updatestatus` to update the status of all my requested observations, I get this error:
```
Traceback (most recent call last):
  File "/var/www/saguaro_tom/manage.py", line 22, in <module>
    main()
  File "/var/www/saguaro_tom/manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/__init__.py", line 436, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/base.py", line 412, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.10/dist-packages/django/core/management/base.py", line 458, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.10/dist-packages/tom_observations/management/commands/updatestatus.py", line 46, in handle
    failed_records[facility_name] = instance.update_all_observation_statuses(target=target)
AttributeError: 'SwiftFacility' object has no attribute 'update_all_observation_statuses'
```
Is the Swift ToO API able to retrieve observation statuses after they are requested? If so, it would be great to add this functionality. If not, we should at least make this not crash.

### PR DESCRIPTION
**Pull request by phycodurus**
the commits tell the tale :-)

Closes #10

### CODE REVIEW

**Review by jchate6 — CHANGES_REQUESTED** [review]
Is the 500s exposure time default from somewhere?
Is this a reasonable exposure time for a swift observation?

The monitoring frequency might benefit from additional help text. I think it should be more explicit if `monitoring_freq` = 2 and `monitoring_units`=hour means once every 2 hours or twice every hour.

File: tom_swift/swift.py [hunk 1]
```diff
@@ -179,20 +197,28 @@ class SwiftObservationForm(BaseObservationForm):
...
-        label='UVOT filter mode',
-        initial='0x9999',
-        help_text=mark_safe(('Supply a specific UVOT Filter mode or written instructions.'
-                             ' See <a target=_blank'
-                             ' href=https://www.swift.psu.edu/operations/mode_lookup.php>'
-                             'UVOT Mode Lookup Tool</a>'
+        label='UVOT Filter Mode',
+        initial=SWIFT_UVOT_FILTER_MODE_CHOICES[-2],
+        choices=SWIFT_UVOT_FILTER_MODE_CHOICES,
+        help_text=mark_safe((f'These are some common UVOT Filter modes.'
```
[the remark this answers is not in the document]
Comment (phycodurus) [reply tom_swift/swift.py:210]: User may want to provide UVOT filter mode instructions even when using 0x9999. (Also, we may have more input from the Swift Ops team on how UVOT filter mode presented).

**Comment by jchate6** [comment]
@phycodurus Before release, should we remove the alpha release warning in the readme?

**Comment by jchate6** [comment]
Also you need to update the dependencies to require `tomtoolkit = "^2.15"`

File: tom_swift/swift.py [hunk 2]
```diff
@@ -127,15 +143,17 @@ class SwiftObservationForm(BaseObservationForm):
...
         widget=forms.Textarea(attrs={
             'rows': 8,
-            'placeholder': 'A persuasive paragraph or two explaining why this object requires rapid observation.'})
-        )
+            'placeholder': 'Two or three persuasive paragraphs explaining why this object requires rapid observation.'
+        })
+    )
 
     #
     # Exposure requested time (total)
```
[the remark this answers is not in the document]
Comment (phycodurus) [reply tom_swift/swift.py:159]: yes

File: tom_swift/swift.py [hunk 3]
```diff
@@ -127,22 +143,27 @@ class SwiftObservationForm(BaseObservationForm):
...
     #
-    exposure = forms.FloatField(required=False, label='Exposure time requested [s]', initial=500)
+    exposure = forms.FloatField(
+        required=True,
```
Comment (jchate6) [inline tom_swift/swift.py:162]: I think there's an issue here when this field is hidden and blank that causes an error when trying to validate:
`An invalid form control with name='exposure' is not focusable. <input type=​"number" name=​"exposure" value=​"12" step=​"any" class=​"numberinput form-control" required id=​"id_exposure">​
`

File: tom_swift/swift.py [hunk 4]
```diff
@@ -127,22 +143,27 @@ class SwiftObservationForm(BaseObservationForm):
...
+        label='Total exposure time requested [s]')
     exp_time_just = forms.CharField(
-        required=False, label='Exposure Time Justification',
+        required=True,
```
Comment (jchate6) [inline tom_swift/swift.py:165]: This has the same issue as Total Exposure time above.

**Review by jchate6 — CHANGES_REQUESTED** [review]
Once I have values in the exposure time and exposure justification fields, I can hit validate and I get the following error: `Could not decode JWT, username/secret wrong?`

I'm pretty sure my swift credentials are accurate though.

File: tom_swift/swift.py (hunk 3 above)
Comment (jchate6) [reply tom_swift/swift.py:162]: This error is invisible unless the console is open and just looks like nothing happens when you try to validate.

**Comment by jchate6** [comment]
I'm experiencing the same `Could not decode JWT, username/secret wrong?` Is the Username different from the one I use to log into www.swift.psu.edu?

**Comment by jchate6** [comment]
I have even tried making a new set of credentials.

**Review by jchate6 — APPROVED** [review]
Figured it out. I was still using `SWIFT_PASSWORD` from an earlier version.

**Pull request merged**